Automattic/crowdsignal-ui
# add permalink handling to editor

Note, needs D86379 for API changes both must be applied to sandbox for testing.

Overview:
This change brings our project section of the editor sidebar closer in alignment with the native WordPress.com post/page editor. The Status & Visibility and permalink sections were combined into a single section called Summary.

![Screen Shot 2022-08-25 at 1 25 48 PM](https://user-images.githubusercontent.com/12895386/186730468-1db1d322-fd22-4e86-ad84-cc360ea1391e.png)

A slight update was made to Visibility to add a header with a close button to match WordPress.com and the same header was added to the new URL section.

The URL section is the main attraction, of course.  It features the new ability for the user to update the project's slug.  

![Screen Shot 2022-08-25 at 1 27 44 PM](https://user-images.githubusercontent.com/12895386/186730809-cc1335f0-807e-4670-a220-0c89744559a4.png)

In addition, the API was updated to both allow for slug editing as well as a new behavior.  The backend will only change the slug to the tile in the event there is no slug set already.  Once the slug is set (whether it's by adding a title, or manually setting a slug) the back end will not attempt to change the slug unless a new slug is set by the user.

Testing instructions:
Be sure to also apply D86379 for the necessary API changes.

Create a new project.  The URL button in the project sidebar will be disabled until the project can be published (a first draft is saved).  Save a first draft, but do not change the title.

Ensure you can click on the URL to open the permalink dialogue.  If a title was not set, you will see the permalink field is blank.  Don't make any changes to the permalink and update the project's title and save the project.  Check the permalink and verify it's set to the title now.

Change the title and save again, verify the permalink has not changed.

Now change the permalink.  Notice the message changes under the text field to let the user know they need to save the project to change the permalink.  Save the project and then verify it indeed has the new permalink.
